addons/mail/static/src/env/test_env.js

```diff
@@ -0,0 +1,153 @@
+odoo.define('mail/static/src/env/test_env.js', function (require) {
+'use strict';
+
+const ModelManager = require('mail/static/src/model/model_manager.js');
+const { makeDeferred } = require('mail/static/src/utils/deferred/deferred.js');
+const { nextTick } = require('mail/static/src/utils/utils.js');
+
+const { Store } = owl;
+const { EventBus } = owl.core;
+
+/**
+ * @param {Object} [providedEnv={}]
+ * @returns {Object}
+ */
+function addMessagingToEnv(providedEnv = {}) {
+
+    const defaultEnv = Object.assign({}, providedEnv, {
+        session: Object.assign({
+            is_bound: Promise.resolve(),
+            name: 'Admin',
+            partner_display_name: 'Your Company, Admin',
+            partner_id: 3,
+            uid: 2,
+            url: s => s,
+        }, providedEnv.session)
+    });
+
+    let env = Object.assign({}, defaultEnv);
+
+    /**
+     * Messaging store
+     */
+    const store = new Store({
+        env,
+        state: {},
+    });
+
+    /**
+     * Registry of models.
+     */
+    env.models = {};
+    /**
+     * Environment keys used in messaging.
+     */
+    Object.assign(env, {
+        autofetchPartnerImStatus: false,
+        destroyMessaging() {
+            if (env.modelManager) {
+                env.modelManager.deleteAll();
+                env.messaging = undefined;
+            }
+        },
+        disableAnimation: true,
+        isMessagingInitialized() {
+            if (!this.messaging) {
+                return false;
+            }
+            return this.messaging.isInitialized;
+        },
+        loadingBaseDelayDuration: 0,
+        messaging: undefined,
```

Comment:
I hesitated to put all of them in `mail/static/src/utils/test_utils.js`, but I prefer to keep them in separate file so that it kinda follows `web.test_env` (other contributors of Odoo repository will likely open this file when looking at `web/test_env.js`, which is a good thing!), and it's quite controversial for us to keep it away from our `mail/test_utils.js` file (also it reduces complexity/scope of `mail/test_utils.js`, which is a nice thing as file becomes big right now).

Also having `env.js` and `test_env.js` motivates having a folder `env`... Yeah, that may be a weak argument, but I took that into account.